index.bs

```diff
@@ -4688,24 +4682,8 @@ partial interface MLGraphBuilder {
                 1. Let |inputWidth| be |inputShape|[2].
                 1. Let |channels| be |inputShape|[3].
         </dl>
-    1. If |outputSizes| is not given, then:
```

Comment:
`outputSizes` is optional, we should keep this checking. I feel the existing code has an issue, it should use `outputSizes` if it is given?

Reply:
Yeah, the previous `if` looked like a typo (so that should be fixed in any case).

Well, with this change, `outputSizes` would be required (caller is explicit now). Do you think this simplification to be problematic? 🤔

Reply:
Then `outputSizes` are not an option anymore, should it be moved to another parameter? Like

```js
 MLOperand averagePool2d(MLOperand input, sequence<[EnforceRange] unsigned long> outputSizes, optional MLPool2dOptions options = {});
```

Reply:
https://w3ctag.github.io/design-principles/#prefer-dictionaries suggests:

> You should also consider accepting mandatory parameters through a dictionary, if it would make the API more readable, especially when they are of primitive types.

...but it also says:

> The dictionary itself should be an optional argument, so that if the author is happy with all of the default options, they can avoid passing an extra argument.

I'd err on the side of putting required arguments into the dictionary - and making the dictionary itself required - because:
- It's more readable, since primitive types are now described (as the Design Principles suggest)
  - `averagePool2d(input, {outputSizes: [2, 3]})` > `averagePool2d(input, [2, 3])`
- It's more readable, since it's otherwise not obvious to the caller why some configuration options belong in the options dictionary while others don't
  - `averagePool2d(input, {outputSizes: [2, 3], layout: "nchw"})` > `averagePool2d(input, [2, 3], {layout: "nchw"})`
- It's more extensible, since the options dict can be changed without changing the method signature (though making a non-required dictionary field required would still be a breaking change)

Reply:
> I'd err on the side of putting required arguments into the dictionary - and making the dictionary itself required

SGTM. Thanks for the explanation, @a-sully !

Reply:
Will do.

Also I just reread this bit "When the output sizes are explicitly specified, the output padding values in [outputPadding](https://www.w3.org/TR/webnn/#dom-mlconvtranspose2doptions-outputpadding) are ignored" which implies that since outputSizes is required now, that outputPadding is always ignored, meaning we can delete it.

Reply:
> and making the dictionary itself required

Hmm, the following didn't work (bikeshed errors):

```
  MLOperand averagePool2d(MLOperand input, required MLPool2dOptions options);
```